fastlane/lib/fastlane/actions/xcodes.rb

```diff
@@ -0,0 +1,150 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      XCODES_XCODE_PATH = :XCODES_XCODE_PATH
+    end
+
+    class XcodesAction < Action
+      def self.run(params)
+        # Provide xcodes with the necessary env vars
+        ENV["XCODES_USERNAME"] = params[:username]
+        ENV["XCODES_PASSWORD"] = params[:password]
+        binary = params[:binary_path]
+        select_for_current_build_only = params[:select_for_current_build_only]
+        version = params[:version]
+        if params[:update_list] && !select_for_current_build_only
```

Comment:
What would be the issue if we allow updating the list even if we want to select it for the current build only?

Reply:
Prior to release v1.0.0, updating the list required Apple authentication, so since update_list is true by default, if you just wanted to pass `select_for_current_build_only: true`, you wouldn't want to be bothered to authenticate with Apple. Now it doesn't require authentication anymore, so I don't see any issues with that!

@MattKiazyk could you confirm what is the exact behavior of the new auth-less requests in v1.0.0? Is it true that there are zero chances of requiring an authentication, regardless of the flow, regardless of which Xcode versions are already installed, macOS version, etc?

Reply:
If no user/pass is required EVER ☝️ then I can go ahead and remove all the user/pass management completely from this new action ✨

Reply:
@rogerluan  On Xcodes - if using the xcoderelease.com api (default) it is not required on the latest version. If using the Apple source it still would be.

Reply:
Thanks @MattKiazyk ! 🙏

Reply:
I just removed the requirement to provide xcodes action Apple user/pass 💪